src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -242,11 +250,23 @@ internal static SyntaxToken RescanInterpolatedString(InterpolatedStringExpressio
                 interpolatedString.GetLastToken().GetTrailingTrivia());
         }
 
+        internal enum InterpolatedStringKind
+        {
+            /// <summary>
+            /// Normal interpolated string that just starts with $"
+            /// </summary>
+            Normal,
+            /// <summary>
+            /// Verbatim interpolated string that starts with $@" or @$"
+            /// </summary>
+            Verbatim,
+        }
+
         [NonCopyable]
-        private struct InterpolatedStringScanner
+        private ref struct InterpolatedStringScanner
```

Comment:
It wasn't clear to me why this is now a ref struct.

Reply:
basically, it's intended to only live on the stack as a temp.  i meant to do this when i originally added the NonCopyable.  but some bad merges seems to have lost htat.

Reply:
That's fine. It might make sense to add a remark that the purpose of the ref is to enforce its temporary noncopyable nature rather than to enable it to contain embedded references.

Reply:
sure.  will do!